### ISSUE TITLE
What about using HaikuOS as base system?

### ISSUE THREAD

**Issue by CyanoMox** [helloSystem/hello#52]
Hello there,
I introduce myself by saying I am pretty new to OS development, and then I know I'm not getting all the things right.

I've read most of the content in this github issues section and guidelines for this project. Let me say that I'm impressed by the direction that this system wants to follow. I was imagining something like that on my own, and in this regard, I have to highlight some additional ideas I have, but I think I'll do it in another issue, because I don't want to bloat this one.

So, what is the point here?

I think that the idea of choosing FreeBSD over Linux is brilliant because it avoids all the deep inconsistencies between Distros and Base-Distros that make any Linux-Based OS so cumbersome and tedious in desktop usage.
But I also read that FreeBSD doesn't target desktop as its main usage case. I've read about rc.d problems and huge memory usage by some modules - which is not very much necessary for what we need -, we should integrate Xorg or using just the framebuffer and avoid graphics server completely, we have to reconfigure a lot of stuff, and so on.
I also was observing the Haiku project for a while (a copule of years), and since now they put out beta 2, I decided to try it out on a bare-metal installation (no VM), and I found it very polished and insanely fast!
Most of the operations we are in order to do on FreeBSD are just in the right way there in Haiku, so we would need less work and have a more coherent base OS below, which is specifically targeting desktop systems and professional workstations.

Most of the problems mentioned by the usage of FreeBSD should be resolved quite automatically where using HaikuOS as core system.

Haiku, at the actual stage, has only few software libs and applications ported, but since it is a fully-POSIX-compliant system, ports can be done easily, and Haiku itself offers a nice and automatized porting application for that, which is called `haikuporter`. With the diffusion of the OS, I have no doubt that most applications would be ported by developers themselves to that System. This is also true on BSD, so no disadvantages here. Same on drivers, that instead need to be rewritten, just like on FreeBSD, though. Both Haiku and FreeBSD have very few drivers and there is not much we can do for this other than make these systems spread across Open Source Community and "Mere Mortals" Userbase. Things will get rounded off only then on both Systems.

Now, I have heard that Haiku devs don't want Distros on their OS, but their kernel and Base System is still Open Source and there must be some kind of acceptable compromise. In the end, we would only make some reskin of their "DE" (I don't know if that could be called a DE, btw) and introduce some kind of GNUstep-ish app allocation scheme.
I think that they are tied up to BeOS too much, because they want backwards compatibility and "look and feel", but IMHO that's only an exercise in style with no benefits whatsoever in practical usage nowadays.
In case they will not permit us to use their entire OS, we should only take their kernel and start from there.

Sure, some research must be done so know if the Haiku kernel itself could be more suitable that BSD kernel. I was setting up some kernel benchmarks on my own between Linux and FreeBSD kernel, and I wanted to add Haiku as well. I was using `phoronix-test-suite benchmak kernel` command, but neither phoronix nor libraries and tests were compiled for Haiku up to this time.

Let me know what you think, guys!

Thank you for reading this.
-Cyano

**Comment by probonopd** [comment]
Hello @CyanoMox. Indeed I have tested Haiku OS and I like it very much.

In fact, I have written a whole article series on it:
https://medium.com/@probonopd/my-first-day-with-haiku-shockingly-good-8930cad4bbb0

While it has _many_ advantages, I chose FreeBSD as the base for helloSystem because of
* Size of the developer community
* Linux application compatibility
* Availability of 3D acceleration including proprietary drivers
* Availability of mainstream web browsers with good performance
among others.

That being said, I think Haiku OS has a lot going for it and I hope it will gain traction. I think it is excellent at what it is, we can all learn a lot from it, but I don't think I can contribute much to it besides testing since they seem to be settled on BeOS like user experience.

**Comment by CyanoMox** [comment]
I understand your points.
Do you think that FreeBSD could become as effective and efficient as Haiku does with some hard work? For now I hardly have seen anything similar either in the most lean Linux distros.

For 3D acceleration, I knew that FreeBSD had a very poor driver support from both hw manufaturers and opensource community. Are you sure that the situation is better in BSD than in Haiku?
For Browsers: porting are needed to Haiku, and they will come anyway. I don't think this to be a problem in a long run, considering they are quickly approaching Beta 3.

Anyway, I'll support this project as I can :D

**Comment by probonopd** [comment]
> Are you sure that the situation is better in BSD than in Haiku?

Yes:

![image](https://user-images.githubusercontent.com/2480569/100665554-87001400-3358-11eb-9dd4-ac8f3ae7d125.png)

> For Browsers: porting are needed to Haiku, and they will come anyway.

Hopefully. Looking forward to seeing Chromium.

The work of the Haiku team is amazing and I will for sure follow it.

**Comment by grahamperrin** [comment]
Things to love about Haiku: many. 

For me, one feature is visually outstanding: the **title bars**, as pictured at https://www.haiku-os.org/slideshows/haiku-1/

Rationale: mainly the _variable width_, as described at [BeOS-like title bars with KDE Plasma Desktop 5 : kde](https://www.reddit.com/r/kde/comments/8hfa3b/beoslike_title_bars_with_kde_plasma_desktop_5/)

**Comment by probonopd** [comment]
> Things to love about Haiku: many.

Indeed! It's just that they are building a system that's appealing to switchers from BeOS, whereas this project is about creating a system that's appealing to switchers from the Mac. But I agree that under the hood one could share _a lot_, and Haiku OS is architected _way_ better than the typical "Linux" (XDG) desktop in my opinion.
